localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/aws.py

```diff
@@ -388,14 +393,12 @@ def invoke(self, context: RestApiInvocationContext) -> Response:
         headers.update(response_headers)
         # TODO: look into remapped headers (like HTTP_PROXY and integration response handlers)
 
-        response = Response(
+        return EndpointResponse(
             headers=headers,
-            response=lambda_response.get("body") or "",
-            status=lambda_response.get("statusCode") or 200,
+            body=to_bytes(lambda_response.get("body") or ""),
+            status_code=lambda_response.get("statusCode") or 200,
```

Comment:
Question: Can the status code of a lambda response be a string?

Reply:
Good question, I'll try to quickly check with a test. If it cannot, this should probably be added to the validation of the lambda response as well. Will update 👍

Reply:
It can yes, and it's allowed:
```
Endpoint response body before transformations: {"statusCode": "201"}
```

Updating the code now!